### DISCUSSION TITLE
<Question>Output Console.WriteLine

### DISCUSSION THREAD

**Discussion by H1Gdev** [dotnet/winforms#4568] [Q&A] [answered]
* .NET Core Version: 

netcoreapp3.1

* Have you experienced this same bug with .NET Framework?: 

No

**Problem description:**

WindowsFormsApp1.csproj
```csproj
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <UseWindowsForms>true</UseWindowsForms>
  </PropertyGroup>

</Project>
```

Form1.cs
```cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace WindowsFormsApp2
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            Console.WriteLine("TEST");    // Add only this line.
        }

    }
}
```

**Expected behavior:**

- `TEST` is output to `OUTPUT`. (Like .Net Framework Forms app)

Please tell me how to output string to `OUTPUT`.

**Minimal repro:**

1. use Microsoft Visual Studio Community 2019 Version 16.8.5
1. Create `Windows Forms App(.NET)`
1. Add only `Console.WriteLine("TEST");`

**Comment by RussKie** [comment, marked as answer]
A console window is not shown for UI app by default. To show the console set the `DisableWinExeOutputInference` property to `true` in your project file.
```xml
<DisableWinExeOutputInference>true</DisableWinExeOutputInference>
```

 [This doc](https://docs.microsoft.com/dotnet/core/compatibility/windows-forms/5.0/automatically-infer-winexe-output-type) describes a workaround, and whilst it targets .NET 5.0, it also fixes the issue for .NET Core 3.1 and earlier.

**Reply by H1Gdev** [reply]
Thanks.

After changing \<OutputType\> and \<DisableWinExeOutputInference\>, console is shown.

WindowsFormsApp1.csproj
```csproj
<Project Sdk="Microsoft.NET.Sdk.WindowsDesktop">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp3.1</TargetFramework>
    <UseWindowsForms>true</UseWindowsForms>
    <DisableWinExeOutputInference>true</DisableWinExeOutputInference>
  </PropertyGroup>

</Project>
```

But I want to show it in `OUTPUT` window of Visual Studio.(Like .Net Framework Forms app)
Do you know how to do this ?

**Reply by RussKie** [reply]
To show in VS please use `Debug.WriteLine`.

**Reply by H1Gdev** [reply]
Thanks !!
I will use `Debug.WriteLine`.
